src/lparser.c

```diff
@@ -771,6 +771,39 @@ static void close_func (LexState *ls) {
   luaM_shrinkvector(L, f->upvalues, f->sizeupvalues, fs->nups, Upvaldesc);
   ls->fs = fs->prev;
   luaC_checkGC(L);
+
+#ifdef USE_YK
+  /*
+   * Identify loops in the Lua program and insert yk locations there.
+   *
+   * We do this here (and not in, e.g. `luaK_code()`), to ensure that bytecode
+   * operands (required to find loop starts) have been finalised.
+   */
+  f->yklocs = luaM_newvectorchecked(L, fs->pc, YkLocation);
+  for (int pc = 0; pc < fs->pc; pc++) {
+    Instruction i = f->code[pc];
+    f->yklocs[pc] = yk_location_null();
+    /*
+     * The computations for finding the start of loops is derived from
+     * `PrintCode()` in `luac.c`. Note that we have to deduct one because luac
+     * prints bytecode pcs starting from 1.
+     */
+    int loop_pc = INT_MAX;
```

Comment:
How about something like:

```c
int loop_pc;
if ((GET_OPCODE(i) == OP_JMP) && (GETARG_sJ(i) < 0)) {
    loop_pc = GETARG_sJ(i) + pc + 2 - 1;
} else if (GET_OPCODE(i) == OP_FORLOOP) {
    loop_pc = pc - GETARG_Bx(i) + 2 - 1;
} else { return; }
f->yklocs[loop_pc] = yk_location_new();
```

to avoid the INT_MAX edge case.

Reply:
That's fine if we don't want the assert.

Reply:
The INT_MAX aspect is probably incorrect on some platforms, so foregoing the assert seems OK.

Reply:
Hopefully [f0b504b](https://github.com/ykjit/yklua/pull/111/commits/f0b504b3972653bb33bd559bf18f3e28a890e116) is a decent compromise.

I considered doing 2 loops:
 - one to initialise every pc with a null location
 - a second to overwrite nulls with non-null locations where necessary

This would be more fool-proof, but less performant.